### PR TITLE
[RUSH-1675][WIP] Add OTKit-icons token and icon generation pipeline

### PR DESCRIPTION
**Pull request by karuto**
### Overview
This PR is a result of the SVG icons [discussion](https://github.com/opentable/design-tokens/issues/93) - let's introduce a designer-friendly solution to standardize our SVG icon usages.

Here's my proposed process:
 1. Drop any normal SVG files, like `ic_information.svg`, into the `assets/` folder.
 2. Type `npm run build-icons` to trigger the script to build the `OTKit/otkit-icons/token.yml` automatically. **The file name will become the key, and the optimized / standardized SVG content will be the value.**
3. Publish the token as normal. Our standard `theo` pipeline will publish it as a `common.js` format file.
4. Developer can use this token in a JavaScript project like so:

```javascript
import React from 'react';
import icons from 'otkit-icons/token.common.js';

export default class RandomComponent extends React.PureComponent {

  render() {
    return (
      <div>
      {icons.ic_information} // here's the icon
      </div>
    );
  };

}
```

**NOTE: Still work in progress. Features I plan to add:**
- [x] Pick a node yaml generator to automate the yaml generation
- [ ] Adjust svgo so that it removes outer SVG tag (but then we have to build the component wrapper, and the example above won't work anymore, so maybe not for V1)
- [ ] Adjust theo so that it only generates common.js for this token (do we ever need CSS version of these icons' paths?)
- [ ] Perhaps make the icon generator run automatically during pre-commit (not sure if we even want this, maybe not for V1)

### CODE REVIEW

File: OTKit/otkit-icons/token.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+props:
+  ic_information:
+    value: >-
```
Comment (karuto) [inline OTKit/otkit-icons/token.yml]: Not sure where this `>-` coming from... need to debug 😭

**Comment by matteofigus** [comment]
I like the idea thanks for exploring this @karuto.
Out of curiosity, do you have opinions about exporting only the paths VS the whole xml? I am trying to understand pros/cons of both.

**Comment by ericz82** [comment]
Question: when a developer uses an icon token in a React project, will the syntax
```
render() {
    return (
      <div>
      {icons.ic_information} // here's the icon
      </div>
    );
  };
``` 
actually work or will `dangerouslySetInnerHTML` be required since the SVG raw is in XML syntax which would be escaped?

**Comment by karuto** [comment]
@ericz82 It should just work, because the SVG markup is still in react land.

**Comment by karuto** [comment]
@lastquestion 

> I wonder if we should be moving the work, logic, etc. from root into the actual otkit-icons. The script, etc. belong to the token repo, not the monorepo as a whole.

I'm ok with this, the location of the script (and the assets folder) is by no means finalized, is very portable.

> Should we actually save the original svg assets in this repository at all? shock.

Yeah I think that's what @matteofigus and @nickbalestra agreed upon, so that designers can easily add / modify. They won't be exposed to consumers of the token anyway.

> I would say that we should only store the original SVG assets here if design wants to use this repo AND that they use these SVG assets as the file they edit.

Yes that's the intention. Feel free to chime in @francesyun.

> If we do store the SVG here, then perhaps we don't need to check in the YML file at all? It should happen as part of pre-publish, but it doesn't need to be checked in.

Good idea, you are right that the yaml is really a middle step and we only consume the common.js output in the end, I haven't explored on messing around with theo build steps yet.

**Comment by karuto** [comment]
@matteofigus

> Out of curiosity, do you have opinions about exporting only the paths VS the whole xml? I am trying to understand pros/cons of both.

Yes I do, there are pros and cons for each, and we can potentially support both. I will reply you tomorrow, it's pretty late here in UK 😭

**Comment by lastquestion** [comment]
@karuto, @ericz82 : @ericz82 is correct. https://codepen.io/anon/pen/KQXyWa?editors=1111

**Comment by karuto** [comment]
@lastquestion Thanks for the snippet, in that case I suggest we can set innerHTML in the icon component in ot-react-ui, so that end consumers won't need to worry about it and will just use the composited component directly.

**Comment by francesyun** [comment]
> I would say that we should only store the original SVG assets here if design wants to use this repo AND that they use these SVG assets as the file they edit.
Yes that's the intention. Feel free to chime in @francesyun.

Yes that sounds right.

**Comment by lastquestion** [comment]
Closed in favor of #135

**Pull request closed without merging**